apps/scan/backend/src/scanners/custom/state_machine.ts

```diff
@@ -0,0 +1,1303 @@
+          states: {
+            resetting: {
+              invoke: {
+                src: reset,
+                onDone: {
+                  target: 'disconnect_after_reset',
+                },
+                onError: {
+                  target: 'disconnect_after_reset',
+                },
+              },
+            },
+            disconnect_after_reset: {
+              initial: 'waiting_to_retry_connecting',
+              states: {
+                waiting_to_retry_connecting: {
+                  after: { DELAY_RECONNECT: 'reconnecting' },
+                },
+                reconnecting: {
+                  invoke: {
+                    src: connectToCustom(createCustomClient),
+                    onDone: {
+                      target: '#checking_initial_paper_status',
+                      actions: assign({
+                        client: (_context, event) => event.data,
+                        error: undefined,
+                      }),
+                    },
+                    onError: 'waiting_to_retry_connecting',
+                  },
+                },
+              },
+            },
+          },
+        },
+        both_sides_have_paper_while_scanning: {
+          entry: clearError,
+          invoke: pollPaperStatus,
+          on: {
+            SCANNER_BOTH_SIDES_HAVE_PAPER: doNothing,
+            // Sometimes we get a no_paper blip when removing the front paper quickly
+            SCANNER_NO_PAPER: doNothing,
+            // After the front paper is removed:
+            SCANNER_READY_TO_EJECT: [
+              // If we already interpreted the paper, go back to routing to the
+              // resulting state for that interpretation
+              {
+                target: 'interpreting.routing_result',
+                cond: (context) => context.interpretation !== undefined,
+              },
+              // Else, if we have scanned images, try to interpret them
+              {
+                target: 'interpreting.starting',
+                cond: (context) => context.scannedSheet !== undefined,
+              },
+              // Otherwise, reject the back paper
+              {
+                target: 'rejecting',
+                actions: assign({
+                  error: new PrecinctScannerError('both_sides_have_paper'),
+                }),
+              },
+            ],
+          },
+        },
+        // If we see an unexpected error, try disconnecting from the scanner and
+        // starting over.
+        error: {
+          id: 'error',
+          initial: 'disconnecting',
+          states: {
+            // First, try disconnecting the "nice" way
+            disconnecting: {
+              invoke: {
+                src: closeCustomClient,
+                onDone: 'cooling_off',
+                onError: 'unexpected_error',
+              },
+              after: {
+                DELAY_RECONNECT_ON_UNEXPECTED_ERROR: '#disconnected',
```

Comment:
I think I used this delay here as it seemed close to being the right thing, but maybe this should be a different more specific delay as its not really an unexpected error

Reply:
This seems okay to me, but I'd be happy to change it if you have a suggestion for a named delay that's clearer.